src/Data/Audience.elm

```diff
@@ -35,6 +42,129 @@ type alias Audience =
     }
 
 
+{-| Information of an `AudienceType` variant
+-}
+type alias AudienceTypeInfo msg =
+    { index : Int
+    , text : String
+    , icon : Html.Html msg
+    }
+
+
+
+-- Decoders
+
+
+{-| `Result` of the JSON parsing.
+It returns `Ok ( List Audience )` if the decoding was successful.
+It returns `Err ( ErrorCode, ErrorDesc )` if the decoding failed.
+-}
+audiences : Result ( String, String ) (List Audience)
+audiences =
+    Result.mapError
+        (\decoderError -> ( "JSON-A", "Parse audiencesJSON says: " ++ Json.Decode.errorToString decoderError ))
+        (Json.Decode.decodeString audiencesDecoder audiencesJSON)
+
+
+audiencesDecoder : Json.Decode.Decoder (List Audience)
+audiencesDecoder =
+    Json.Decode.field "data" (Json.Decode.list audienceDecoder)
+
+
+audienceDecoder : Json.Decode.Decoder Audience
+audienceDecoder =
+    Json.Decode.map4 Audience
+        (Json.Decode.field "id" Json.Decode.int)
+        (Json.Decode.field "name" Json.Decode.string)
+        (Json.Decode.field "type" audienceTypeDecoder)
+        (Json.Decode.field "folder" (Json.Decode.nullable Json.Decode.int))
+
+
+audienceTypeParser : String -> Result String AudienceType
+audienceTypeParser string =
+    case String.toLower string of
+        "user" ->
+            Ok Authored
+
+        "shared" ->
+            Ok Shared
+
+        "curated" ->
+            Ok Curated
+
+        unknown ->
```

Comment:
Wow, love the usage of `unknown` instead of `_` and having the problem in the Error. Nice thinking. 👍